sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -2548,6 +2553,10 @@ class Dataset[T] private[sql](
    * the state. In addition, too late data older than watermark will be dropped to avoid any
    * possibility of duplicates.
    *
+   * Note that for a streaming [[Dataset]], this method only returns distinct rows only once
+   * regardless of the output mode, which the behavior may not be same with using distinct in
+   * SQL statement against streaming [[Dataset]].
```

Comment:
ditto, there is no `dropDuplicates` in SQL. If we want to highlight differences between SQL API and Dataset API, let's update the doc of `Dataset.distinct`